File: 1657-determine-if-two-strings-are-close/1657-determine-if-two-strings-are-close.java
class Solution {
    public boolean closeStrings(String word1, String word2) {
        int[] letter1 = new int[26];
        int[] letter2 = new int[26];

        for (int i = 0; i < word1.length(); i++) {
            letter1[word1.charAt(i) - 'a']++;
        }

        for (int i = 0; i < word2.length(); i++) {
            letter2[word2.charAt(i) - 'a']++;
        }

        for (int i = 0; i < 26; i++) {
            if ((letter1[i] != 0 && letter2[i] == 0) ||
                (letter1[i] == 0 && letter2[i] != 0)) {
                return false;
            }
        }

        Arrays.sort(letter1);
        Arrays.sort(letter2);

        for (int i = 25; i > -1; i--) {
            if (letter1[i] != letter2[i]) return false;
            else if (letter1[i] == 0) return true;
        }

        return true;
    }
}

File: 1657-determine-if-two-strings-are-close/1657-determine-if-two-strings-are-close.py
class Solution:
    def closeStrings(self, word1: str, word2: str) -> bool:
        letter1 = [0 for i in range(26)]
        letter2 = [0 for i in range(26)]

        for letter in word1:
            letter1[ord(letter) - ord('a')] += 1

        for letter in word2:
            letter2[ord(letter) - ord('a')] += 1

        for i in range(26):
            if ((letter1[i] and not letter2[i]) or
                (not letter1[i] and letter2[i])):
                return False

        letter1.sort(reverse=True)
        letter2.sort(reverse=True)

        for i in range(26):
            if letter1[i] != letter2[i]:
                return False
            elif letter1[i] == 0:
                return True

        return True